src/Tasks/WriteCodeFragment.cs

```diff
@@ -221,12 +238,24 @@ private string GenerateCode(out string extension)
                         return null;
                     }
 
-                    attribute.Arguments.Add(orderedParameters[i]);
+                    providedOrderedParameters.Add(orderedParameters[i].Value);
                 }
 
-                foreach (CodeAttributeArgument namedParameter in namedParameters)
+                var attribute = new CodeAttributeDeclaration(new CodeTypeReference(attributeItem.ItemSpec));
+
+                // We might need the type of the attribute if we need to infer the
+                // types of the parameters. Search for it by the given type name,
+                // as well as within the namespaces that we automatically import.
+                Lazy<Type> attributeType = new(
+                    () => Type.GetType(attribute.Name) ?? NamespaceImports.Select(x => Type.GetType($"{x}.{attribute.Name}")).FirstOrDefault(),
+                    System.Threading.LazyThreadSafetyMode.None
+                );
+
+                if (
+                    !AddArguments(attribute, attributeType, providedOrderedParameters, true)
```

Comment:
mind adding the argument name for literals such as `true`?

Reply:
Done. 😄